bool initialized_
bool data